### PR TITLE
GEODE-4030, GEODE-3495: Bump dependencies to new versions

### PR DESCRIPTION
**Pull request by PurelyApplied**
* Jackson     : 2.8.6 -> 2.9.3
* Jetty       : 9.3.10.v20160621 -> 9.4.8.v20171121
* Spring      :
* - Security  : 4.2.1.RELEASE -> 4.2.3.RELEASE
* - TX        : 4.3.6.RELEASE -> 4.3.13.RELEASE
* - Framework : 4.3.6.RELEASE -> 4.3.13.RELEASE
*
* Log4j / slf4j dependency bump currently blocked by LOG4J2-2152.

Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [x] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [x] Is your initial contribution a single, squashed commit?

- [x] Does `gradlew build` run cleanly?

- [ ] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

**Comment by PurelyApplied** [comment]
`SessionReplicationIntegrationJUnitTest` fails when Log4j is bumped.

In the case of a bump to 2.9.x, it fails to [LOG4J2-2055](https://issues.apache.org/jira/browse/LOG4J2-2055).

In the case of a bump to 2.10.x, it fails to [LOG4J2-2152](https://issues.apache.org/jira/browse/LOG4J2-2152).

Log4j / slf4j updating must then either be deferred until these issues are addressed or until the ofending tests can be rewritten.

After some confusing collaboration, this PR subsumed PR#903.

**Comment by PurelyApplied** [comment]
Precheck ran clean excepting one failure from 
```
org.apache.geode.internal.cache.DeltaPropagationDUnitTest > testBug40165ClientReconnects
```

This has been occurring intermittently and is likely unrelated, but I am rerunning the distributed tests in any case.

**Comment by PurelyApplied** [comment]
It also appears that I included some minor test refactorings in this commit.  Changes were largely driven by IDE inspections (`new Callable` to lambdas, etc) and the removal of dead code.

The exception to this is changing how cookies were handled in the `SessionReplicationIntegrationJUnitTest` at the recommendation of @metatype with respect to the Jetty version bump.  This being outside my wheelhouse, I defer to anyone with greater insight into these tests.

File: geode-assembly/src/test/java/org/apache/geode/rest/internal/web/controllers/RestAPIsWithSSLDUnitTest.java [hunk 1]
```diff
@@ -585,21 +622,6 @@ public void testSSLWithTLSProtocol() throws Exception {
     validateConnection(restEndpoint, "TLS", props);
   }
 
-  @Test
-  public void testSSLWithTLSv11Protocol() throws Exception {
```
Comment (jdeppe-pivotal) [inline geode-assembly/src/test/java/org/apache/geode/rest/internal/web/controllers/RestAPIsWithSSLDUnitTest.java]: This test should not be removed. Rather, mark as Ignore and provide Jira

File: geode-assembly/src/test/java/org/apache/geode/rest/internal/web/controllers/RestAPIsWithSSLDUnitTest.java [hunk 2]
```diff
@@ -744,19 +766,6 @@ public void testSSLWithTLSProtocolLegacy() throws Exception {
     validateConnection(restEndpoint, "TLS", props);
   }
 
-  @Test
-  public void testSSLWithTLSv11ProtocolLegacy() throws Exception {
```
Comment (jdeppe-pivotal) [inline geode-assembly/src/test/java/org/apache/geode/rest/internal/web/controllers/RestAPIsWithSSLDUnitTest.java]: Ditto, above

File: geode-assembly/src/test/java/org/apache/geode/rest/internal/web/controllers/RestAPIsWithSSLDUnitTest.java (hunk 1 above)
Comment (metatype) [reply geode-assembly/src/test/java/org/apache/geode/rest/internal/web/controllers/RestAPIsWithSSLDUnitTest.java]: The problem is that all cipher suites for TLSv11 have known vulnerabilities and have been disabled by Jetty.  If we bump the Jetty version we are essentially removing support for TLS v1.1.

Comment (jdeppe-pivotal) [reply geode-assembly/src/test/java/org/apache/geode/rest/internal/web/controllers/RestAPIsWithSSLDUnitTest.java]: Regardless, it's still possible to do it (even if it's bad). Some people will need it for older clients that can only do TLSv1.1.

**Comment by jdeppe-pivotal** [comment]
Merged manually after making some review changes.

**Pull request closed without merging**